### PR TITLE
  Update core API endpoints

### PR DESCRIPTION
**Pull request by daria305**
This PR corresponds to iota-core [PR](https://github.com/iotaledger/iota-core/pull/275).

### CODE REVIEW

File: nodeclient/apimodels/core.go [hunk 1]
```diff
@@ -315,21 +315,28 @@ type (
...
-		LatestSupportedProtocolVersion iotago.Version `serix:"5,mapKey=latestSupportedProtocolVersion"`
+		LatestSupportedProtocolVersion iotago.Version `serix:"6,mapKey=latestSupportedProtocolVersion"`
 	}
 
 	// AccountStakingListResponse defines the response for the staking REST API call.
 	AccountStakingListResponse struct {
-		Stakers []*ValidatorResponse `serix:"0,mapKey=stakers"`
+		Stakers     []*ValidatorResponse `serix:"0,mapKey=stakers"`
+		PageSize    uint32               `serix:"1,mapKey=pageSize"`
+		Coursor     uint32               `serix:"2,mapKey=coursor"`
```
Comment (muXxer) [inline nodeclient/apimodels/core.go]:
```suggestion
		Cursor      uint32               `serix:"2,mapKey=cursor"`
```

Comment (muXxer) [reply nodeclient/apimodels/core.go]: Tbh, I think the cursor needs to be a *string with omitempty... at least that's how it's done in the inx-indexer. :thinking:

File: nodeclient/apimodels/core.go [hunk 2]
```diff
@@ -315,21 +315,28 @@ type (
...
+		RequestedAt iotago.SlotIndex     `serix:"3,mapKey=requestedAt"`
 	}
 
 	// ManaRewardsResponse defines the response for the mana rewards REST API call.
 	ManaRewardsResponse struct {
-		// EpochIndex is the epoch index for which the mana rewards are returned.
-		EpochIndex iotago.EpochIndex `serix:"0,mapKey=epochIndex"`
+		// EpochIndexStart is the starting epoch for range for which the mana rewards are returned.
```
Comment (muXxer) [inline nodeclient/apimodels/core.go]:
```suggestion
		// EpochIndexStart is the starting epoch for the range for which the mana rewards are returned.
```

File: nodeclient/apimodels/core.go [hunk 3]
```diff
@@ -315,21 +315,28 @@ type (
...
+		EpochIndexStart iotago.EpochIndex `serix:"0,mapKey=epochIndexStart"`
+		// EpochIndexEnd is the ending epoch for range for which the mana rewards are returned.
```
Comment (muXxer) [inline nodeclient/apimodels/core.go]:
```suggestion
		// EpochIndexEnd is the ending epoch for the range for which the mana rewards are returned.
```

File: nodeclient/http_api_client.go [hunk 4]
```diff
@@ -35,8 +35,40 @@ const (
 	// GET returns the node info.
 	RouteInfo = "/api/core/v3/info"
 
+	// RouteCongestion is the route for getting congestion details for the account.
+	// GET returns the congestion details for the account.
+	// MIMEApplicationJSON => json.
+	// MIMEVendorIOTASerializer => bytes.
+	RouteCongestion = "/api/core/v3/accounts/%s/congestion"
+
+	// RouteRewards is the route for getting the rewards for staking or delegation based on provieded output.
```
Comment (muXxer) [inline nodeclient/http_api_client.go]:
```suggestion
	// RouteRewards is the route for getting the rewards for staking or delegation based on the provided output.
```

File: nodeclient/http_api_client.go [hunk 5]
```diff
@@ -35,8 +35,40 @@ const (
...
+	// MIMEVendorIOTASerializer => bytes.
+	RouteRewards = "/api/core/v3/rewards/%s"
+
+	// RouteStaking is the route for getting the information about current validators.
+	// GET returns the information about current validators.
+	// MIMEApplicationJSON => json.
+	// MIMEVendorIOTASerializer => bytes.
+	RouteStaking = "/api/core/v3/staking"
+
+	// RouteStakingAccount is the route for getting an account by its accountID.
```
Comment (muXxer) [inline nodeclient/http_api_client.go]: Mhh does that description really match the endpoint?

File: nodeclient/http_api_client.go [hunk 6]
```diff
@@ -35,8 +35,40 @@ const (
...
+	// GET returns the account details.
+	// MIMEApplicationJSON => json.
+	// MIMEVendorIOTASerializer => bytes.
+	RouteStakingAccount = "/api/core/v3/staking/%s"
+
+	// RouteCommittee is the route for getting the information about current committee.
+	// GET returns the information about current committee.
```
Comment (muXxer) [inline nodeclient/http_api_client.go]:
```suggestion
	// RouteCommittee is the route for getting the information about the current committee.
	// GET returns the information about the current committee.
```

File: nodeclient/http_api_client.go [hunk 7]
```diff
@@ -35,8 +35,40 @@ const (
...
+	// GET returns the rewards for the output.
+	// MIMEApplicationJSON => json.
+	// MIMEVendorIOTASerializer => bytes.
+	RouteRewards = "/api/core/v3/rewards/%s"
+
+	// RouteStaking is the route for getting the information about current validators.
+	// GET returns the information about current validators.
+	// MIMEApplicationJSON => json.
+	// MIMEVendorIOTASerializer => bytes.
+	RouteStaking = "/api/core/v3/staking"
```
Comment (muXxer) [inline nodeclient/http_api_client.go]: I'm wondering if `/staking` is a descriptive endpoint, if the response is about the validators :thinking:

File: nodeclient/apimodels/core.go (hunk 1 above)
Comment (daria305) [reply nodeclient/apimodels/core.go:328]: yeah, but you need there string to specify the query, and I simply use index, so I thought int is enough

File: nodeclient/http_api_client.go (hunk 7 above)
Comment (daria305) [reply nodeclient/http_api_client.go]: changed to validators, might be actually better

**Review by muXxer — APPROVED** [review]

File: nodeclient/http_api_client.go [hunk 8]
```diff
@@ -321,6 +355,65 @@ func (client *Client) BlockIssuance(ctx context.Context) (*apimodels.IssuanceBlo
...
+	query := fmt.Sprintf(RouteRewards, hexutil.EncodeHex(outputID[:]))
+	//nolint:bodyclose
+	if _, err := client.Do(ctx, http.MethodGet, query, nil, res); err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (client *Client) Staking(ctx context.Context) (*apimodels.AccountStakingListResponse, error) {
```
Comment (karimodm) [inline nodeclient/http_api_client.go]: We should also include an endpoint for Delegation

**Review by karimodm — CHANGES_REQUESTED** [review]
Looks good! Just add a Delegation endpoint!

File: nodeclient/http_api_client.go (hunk 8 above)
Comment (daria305) [reply nodeclient/http_api_client.go]: oh But the delegation strictly for rewards? then /rewards endpoint covers both staking and delegation.
The staking endpoint was renamed to `validators` it just better suits the purpose which is for user to have a list of possible validators to delegate to.

**Review by karimodm — APPROVED** [review]

**Pull request merged**